kernel/src/lib.rs

```diff
@@ -45,7 +45,11 @@ pub use crate::platform::systick::SysTick;
 pub use crate::platform::{mpu, Chip, Platform};
 pub use crate::platform::{ClockInterface, NoClockControl, NO_CLOCK_CONTROL};
 pub use crate::returncode::ReturnCode;
-pub use crate::sched::Kernel;
+pub use crate::sched::cooperative::{CoopProcessNode, CooperativeSched};
+pub use crate::sched::mlfq::{MLFQProcessNode, MLFQSched};
+pub use crate::sched::priority::PrioritySched;
+pub use crate::sched::round_robin::{RoundRobinProcessNode, RoundRobinSched};
+pub use crate::sched::{Kernel, Scheduler};
```

Comment:
If we didn't have the legacy sched.rs file, I don't think we would go with that name now. I think we should start improving this by creating a `schedulers` (or `scheduler`) directory, and putting the schedulers in there.

Reply:
Using `sched/` makes it so that the individual schedulers are in the same module as `sched.rs`, so they can use private functions defined in that file (e.g. `processes_blocked()`). If we want to move them into a `schedulers/` directory we either have to:

a) change `processes_blocked()`, `do_process()`, and `Kernel.processes` to `pub(crate)` visibility instead of being private or

b) also change `sched.rs` --> `schedulers.rs`

I am fine with a) , as we sort of have to trust everything in the Kernel crate anyway, so I am not that worried about visibility changes. If you prefer b), I would rather wait and do that in a followup PR, because changing the name of `sched.rs` would make this PR very difficult to review.

Reply:
`sched.rs` should have been `kernel.rs` for a while now. Since after this PR it will definitely not do any scheduling, I think it is time to rename it to what it actually holds, the `Kernel` struct. And then it seems to make sense to have schedulers in a schedulers folder. We can make things pubcrate as needed.

Reply:
After this PR is merged, I will submit a PR that renames `sched.rs` to `kernel.rs` and renames the `sched/` folder to `schedulers/`, making visibility changes as needed.